### DISCUSSION TITLE
Getting more detailed artist information

### DISCUSSION THREAD

**Discussion by kihyyo** [Slyyxp/rsack#49] [Q&A] [answered]
Currently gives only information of album.

In web I can see more detailed information of artist and also in app like who she or he is.

I need some help the way getting more detailed information of artist through https://app.genie.co.kr/???/j_ArtistInfo.json I guess.

And more detailed information of album If possible, Thanks

**Comment by Slyyxp** [comment]
When you say more detailed information of album what kind of information are you looking for?

**Reply by kihyyo** [reply]
Thanks for helping me again.
I can get that information already through using unquote(meta['DATA0']['DATA'][0]['ALBUM_DESC'])
Only problem was this include some unnessary str like html tag.
And looks like same when getting get_artist.meta['artist_info']['artist_profile']
I guess I can solve these problem in the end but Can you just give me some recommendation or tips to eliminate these?
Thank  a lot, again

**Reply by Slyyxp** [reply]
If all you want is to remove HTML tags you can use regex.
I won't implement anything to handle this as HTML tags can be used for preservation.

Here's an example snippet of how you could remove them yourself:
```python
import re
from urllib.parse import unquote

from rsack.clients import genie

def clean_html(html: str, compiler: re.Pattern) -> str:
    """Clean html using regex"""
    return re.sub(compiler, '', html) # Return results of re.sub.

if __name__ == "__main__":
    """Note: re.compile should be called outside of function.
            This prevents having to re.compile every time clean_html() is called.
    """ 
    client = genie.Client() # Instantiate client
    client.auth("username", "password") # Authenticate
    meta = client.get_album(82906357) # Retrieve meta
    compiler = re.compile('<.*?>|&([a-z0-9]+|#[0-9]{1,6}|#x[0-9a-f]{1,6});') # This should cover all cases in this API response.
    print(clean_html(unquote(meta['DATA0']['DATA'][0]['ALBUM_DESC']), compiler)) # Output cleaned album description
```

**Comment by Slyyxp** [comment, marked as answer]
cc891840b1ba6fad66e31b10739027dc4b2a67cb I have implemented the endpoint that I think you're after and have renamed the existing `get_artist()` function to `get_artist_albums()`

Here's a quick snippet you can use to test this:
_(Note: This has not yet been pushed to PyPi)_
```python
from rsack.clients import genie

client = genie.Client()
client.auth("username", "password")
meta = client.get_artist(80288928)
print(meta)
```

**Reply by kihyyo** [reply]
fantastic working ! thanks a lot.
